src/conn/conn_prefetch.c

```diff
@@ -216,18 +209,9 @@ __wt_conn_prefetch_clear_tree(WT_SESSION_IMPL *session, bool all)
      * To avoid a race condition, check if this dhandle is currently being operated on by any other
      * prefetch thread. If so, wait for the thread to finish before proceeding with eviction.
      */
-
-    while (i < WT_PREFETCH_THREAD_COUNT) {
-        /*
-         * We use WT_ORDERED_READ here to guarantee read_once semantics on prefetch_dhandle. We
-         * don't require its ordering constraints.
-         */
-        WT_ORDERED_READ(prefetch_dhandle, conn->prefetch_dhandles[i]);
-        if (prefetch_dhandle == dhandle)
-            __wt_sleep(0, 100);
-        else
-            i++;
-    }
+    if (dhandle != NULL)
+        while (((WT_BTREE *)dhandle->handle)->prefetch_busy > 0)
+            __wt_yield();
 
     __wt_spin_unlock(session, &conn->prefetch_lock);
```

Comment:
The wait logic must be outside the unlock. Otherwise, the prefetch threads will wait for the lock to be available to reduce the counter.